src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -230,13 +230,22 @@ private function addFormSection(ArrayNodeDefinition $rootNode, callable $enableI
                     ->{$enableIfStandalone('symfony/form', Form::class)}()
                     ->children()
                         ->arrayNode('csrf_protection')
-                            ->treatFalseLike(['enabled' => false])
-                            ->treatTrueLike(['enabled' => true])
-                            ->treatNullLike(['enabled' => true])
```

Comment:
I would suggest to keep those `treat*Like` methods hand having a `->beforeNormalization()->ifString()->then(fn ($v) => ['enabled' => null, 'header_name' => $v])->end()` instead. To me, it makes things easier to understand to keep the different normalization rules separate, especially when we have a shortcut for booleans and null

Reply:
alright, updated